### ISSUE TITLE
RoBERTa + RACE accuracy might be wrongly calculated

### ISSUE THREAD

**Issue by xuanqing94** [pytorch/fairseq#1565]
Hi, I have a question for RoBERTa authors. When I tried to evaluate RoBERTa on RACE dataset, I met the same problem as #1534 #1181 I am able to reproduce the official result with the following script:

```
MAX_EPOCH=5           # Number of training epochs.
LR=1e-05              # Peak LR for fixed LR scheduler.
NUM_CLASSES=4
MAX_SENTENCES=1       # Batch size per GPU.
UPDATE_FREQ=16         # Accumulate gradients to simulate training on 8 GPUs.
DATA_DIR=./bin
ROBERTA_PATH=checkpoints/checkpoint_best.pt

fairseq-train $DATA_DIR --ddp-backend=no_c10d \
    --restore-file $ROBERTA_PATH \
    --reset-optimizer --reset-dataloader --reset-meters \
    --best-checkpoint-metric accuracy --maximize-best-checkpoint-metric \
    --task sentence_ranking \
    --num-classes $NUM_CLASSES \
    --init-token 0 --separator-token 2 \
    --max-option-length 128 \
    --max-positions 512 \
    --truncate-sequence \
    --arch roberta_large \
    --dropout 0.1 --attention-dropout 0.1 --weight-decay 0.01 \
    --criterion sentence_ranking \
    --optimizer adam --adam-betas '(0.9, 0.98)' --adam-eps 1e-06 \
    --clip-norm 0.0 \
    --lr-scheduler fixed --lr $LR \
    --fp16 --fp16-init-scale 4 --threshold-loss-scale 1 --fp16-scale-window 128 \
    --max-sentences $MAX_SENTENCES \
    --required-batch-size-multiple 1 \
    --update-freq $UPDATE_FREQ \
    --max-epoch $MAX_EPOCH \
    --valid-subset test \  then change this to `test1` after completion
    --save-dir ./checkpoints/RACE \
    --save-interval-updates 1 \
```

The trick here is to replace the dev data with test data and set the evaluation interval to once per iteration. After the first evaluation on test data we can exit the program and repeat this process on test1.

However, there might be an error when calculating the gross accuracy on RACE data, the correct value should be:
```python
# Number of samples: 1436 (Middle) and 3498 (High)
# Accuracy: 86.5% (Middle) and 81.3%(High)
>>> (1436 * 0.865 + 3498 * 0.813) / (1436 + 3498)                                             
0.8281341710579652
```
Which is different from the reported result 83.2% in [paper](https://arxiv.org/pdf/1907.11692.pdf)

**Comment by 644943** [comment]
please show me your RACE Evaluation code.

**Comment by myleott** [comment]
There are a few options:
* During training you can set `--valid-subset valid,test,test1` and it will print results for valid, test and test1 during training.
* Alternatively you can use `fairseq-validate` to evaluate an existing finetuned model on a new dataset. I'll add some instructions for this approach in a bit.
